internal/tool/kubectl.go

```diff
@@ -94,49 +97,80 @@ func NewKubectl(kubeConfig clientcmd.ClientConfig) (*Kubectl, error) {
 	return kubectl, nil
 }
 
-func (k Kubectl) WaitForDeployments(context context.Context, namespace string, selector string) error {
+func (k Kubectl) WaitForWorkloadResources(context context.Context, namespace string, selector string) error {
 	deadline, _ := context.Deadline()
-	unavailableDeployments := []deploymentNotReady{{Name: "none", Unavailable: 1}}
-	getDeploymentsError := ""
-
-	utils.LogInfo(fmt.Sprintf("Start wait for deployments. --timeout time left: %s ", time.Until(deadline).String()))
-
-	for deadline.After(time.Now()) && len(unavailableDeployments) > 0 {
-		unavailableDeployments = []deploymentNotReady{}
-		deployments, err := listDeployments(k, context, namespace, selector)
-		if err != nil {
-			unavailableDeployments = []deploymentNotReady{{Name: "none", Unavailable: 1}}
-			getDeploymentsError = fmt.Sprintf("error getting deployments from namespace %s : %v", namespace, err)
-			utils.LogWarning(getDeploymentsError)
+	unavailableWorkloadResources := []workloadNotReady{{Name: "none", Unavailable: 1}}
+	getWorkloadResourceError := ""
+
+	// Loop until timeout reached or all requested pods are available
+	utils.LogInfo(fmt.Sprintf("Start wait for workloads resources. --timeout time left: %s ", time.Until(deadline).String()))
+	for deadline.After(time.Now()) && len(unavailableWorkloadResources) > 0 {
+		unavailableWorkloadResources = []workloadNotReady{}
+
+		deployments, errDeployments := listDeployments(k, context, namespace, selector)
+		daemonSets, errDaemonSets := listDaemonSets(k, context, namespace, selector)
+		statefulSets, errStatefulSets := listStatefulSets(k, context, namespace, selector)
+
+		// Else/If block will handle errors from API requests for each
+		// resource type or inspect the resources that are successfully returned
+		if errDeployments != nil {
+			unavailableWorkloadResources = []workloadNotReady{{Name: "none", ResourceType: "Deployment", Unavailable: 1}}
+			getWorkloadResourceError = fmt.Sprintf("error getting deployments from namespace %s : %v", namespace, errDeployments)
+			utils.LogWarning(getWorkloadResourceError)
+			time.Sleep(time.Second)
```

Comment:
This was actually left over from what the function previously did. 
I believe the idea is to avoid excessive calls and rate limiting since it'll immediately hit the API again after these failures. Happy to remove it if this is a non-issue but trying to find the default rate limit info first